spec/shard.md

```diff
@@ -0,0 +1,498 @@
+6. If `file_flags & MDB_FILE_FLAG_WITH_METADATA_EXT != 0`: read 1 × `FileMetadataExt`
+7. Repeat from step 2 until bookend found
+
+## 3. CAS Info Section
+
+**Location**: `footer.cas_info_offset` to `footer.footer_offset`
+
+This section contains CAS (Content Addressable Storage) block information. Each CAS Info block represents a xorb by first having a CASChunkSequenceHeader which contains the number of CASChunkSequenceEntries to follow that make up this block.
+
+### CASChunkSequenceHeader
+
+```rust
+struct CASChunkSequenceHeader {
+    cas_hash: Hash,           // 32-byte CAS block hash
+    cas_flags: u32,           // CAS flags
+    num_entries: u32,         // Number of chunks in this CAS block
+    num_bytes_in_cas: u32,    // Total bytes in CAS block
+    num_bytes_on_disk: u32,   // Bytes stored on disk
+}
+```
+
+**Memory Layout**:
+
+```txt
+┌────────────────────────────────────────────────────────────────┬─────────┬─────────┬─────────┬─────────┐
+│                       cas_hash (32 bytes)                      │cas_flags│num_     │num_bytes│num_bytes│
+│                      CAS Block Hash                            │(4 bytes)│entries  │in_cas   │on_disk  │
+│                                                                │         │(4 bytes)│(4 bytes)│(4 bytes)│
+└────────────────────────────────────────────────────────────────┴─────────┴─────────┴─────────┴─────────┘
+0                                                               32        36        40        44        48
+```
+
+### CASChunkSequenceEntry
+
+Every CASChunkSequenceHeader will have a num_entries number field. This number is the number of CASChunkSequenceEntry items that should be deserialized that are associated with the xorb described by this CAS Info block.
+
+```rust
+struct CASChunkSequenceEntry {
+    chunk_hash: Hash,             // 32-byte chunk hash
+    unpacked_segment_bytes: u32,  // Size when unpacked
+    chunk_byte_range_start: u32,  // Start position in CAS block
+    _unused: [u8; 8],             // Reserved space 8 bytes
+}
+```
+
+**Memory Layout**:
+
+```txt
+┌────────────────────────────────────────────────────────────────┬─────────┬─────────┬─────────────────┐
+│                     chunk_hash (32 bytes)                      │chunk_   │unpacked │    _unused      │
+│                        Chunk Hash                              │byte_    │segment_ │   (8 bytes)     │
+│                                                                │range_   │bytes    │                 │
+│                                                                │start    │(4 bytes)│                 │
+│                                                                │(4 bytes)│         │                 │
+└────────────────────────────────────────────────────────────────┴─────────┴─────────┴─────────────────┘
+0                                                               32        36        40               48
+```
+
+### CAS Info Section Layout
```

Comment:
nit: I would prefer we put the layout at the start of the `3. CAS Info Section` as a way to get an overview of the section layout before diving into each of the specific types.